tests/library/test_json.py

```diff
@@ -597,6 +596,76 @@ def test_anyOf_objects(self, target_obj):
         # The actual check
         _generate_and_check(target_obj, schema_obj)
 
+class TestEnum:
+
```

Comment:
Could you add some 'negative' tests, at least similar to the ones I have? If you can see another way of doing that which will actually generate an error (rather than output relying on the vagaries of the PRNG) that would be even better.

Reply:
I have to admit that I'm a little bit confused by the `Mock`-based tests because of the PRNG. Would it be sensible to use `grammar.match` for the negative cases?

Reply:
Very possibly; I overlooked that when constructing my negative test cases. It would be much better to have something which raised an exception when fed input which doesn't match the grammar.

Reply:
Added some grammar.match based tests, but I don't love that it's not explicitly testing that the failure is what I expect the failure to be. I think in a previous iteration, you were consuming the input byte by byte and making an assertion about which byte failed. What steered you away from that approach?

Reply:
Before I was constructing the grammar object directly, not requiring a language model as well. I may have missed a way of doing the byte-by-byte consumption in the new approach.